usefulUtilities/sit/sitClient.js

```diff
@@ -233,6 +236,110 @@
+        }
+
+        requestSitDataTimer = false;
+
+        var entityPosition = Entities.getEntityProperties(_this.entityID, ["position"]).position;
+        var distance = Vec3.distance(entityPosition, Camera.position);
+        if (distance <= MAX_REQUEST_DISTANCE_M) {
+            if (DEBUG) {
+                console.log("sitClient.js: The camera is close enough to this seat! Requesting sit data from sitServer...");
+            }
+            requestSitData(_this.entityID);
+        } else {
+            if (DEBUG) {
+                console.log("sitClient.js: The camera is too far away from this seat! Waiting " + REQUEST_SIT_DATA_TIMEOUT_MS + "ms before trying again...");
+            }
+            requestSitDataTimer = Script.setTimeout(requestSitDataTimerCallback, REQUEST_SIT_DATA_TIMEOUT_MS);
+        }
+    }
+
+
+    function maybeClearRequestSitDataTimeoutTimer() {
+        currentRequestAttempts = 0;
+        if (requestSitDataTimeoutTimer) {
+            Script.clearTimeout(requestSitDataTimeoutTimer);
+            requestSitDataTimeoutTimer = false;
+        }
+    }
+
+
+    var currentRequestAttempts = 0;
+    var MAX_REQUEST_ATTEMPTS = 3;
+    function requestSitDataTimeoutTimerCallback() {
+        requestSitDataTimeoutTimer = false;
+        currentRequestAttempts++;
+        if (currentRequestAttempts <= MAX_REQUEST_ATTEMPTS) {
+            console.log("sitClient.js: Request for sit data timed out for Entity ID: " + _this.entityID +
+                ". Trying again. This will be attempt " + currentRequestAttempts + "/" + MAX_REQUEST_ATTEMPTS);
+            requestSitData(_this.entityID);
+        } else {
+            currentRequestAttempts = 0;
+            console.log("sitClient.js: Request for sit data timed out for Entity ID: " + _this.entityID +
+                ". " + MAX_REQUEST_ATTEMPTS + " requests for data failed. Not trying again.");
+        }
+    }
+
+
+    var requestSitDataTimeoutTimer = false;
+    var REQUEST_SIT_DATA_TIMEOUT_MS = 6000;
+    function requestSitData(id) {
+        if (DEBUG) {
+            console.log("sitClient.js: Requesting sit data...");
+        }
+
+        currentRequestAttempts++;
+        requestSitDataTimeoutTimer = Script.setTimeout(requestSitDataTimeoutTimerCallback, REQUEST_SIT_DATA_TIMEOUT_MS);
+        Entities.callEntityServerMethod(id, "requestSitData", [MyAvatar.sessionUUID]);
+    }
+
+
+    function requestSitDataReply(id, args) {
+        maybeClearRequestSitDataTimeoutTimer();
+        maybeClearRequestSitDataTimer();
+
+        var data;
+        try {
+            data = JSON.parse(args[0]);
+        } catch (e) {
+            console.log("sitClient.js: Error when parsing sit data reply! "+ e);
+            return;
+        }
+        var isOccupied = data.isOccupied;
+
+        if (DEBUG) {
+            console.log("sitClient.js: Got sit data reply! `isOccupied`: " + isOccupied);
+        }
+
```

Comment:
Could we wait to clear the timeout until we know the data received is valid (no parsing errors)?

Reply:
No - I'd to clear the timeouts immediately so that we stop requesting any data. If we have issues parsing the data reply, something is very wrong, and we don't want to keep trying.